src/lib.rs

```diff
@@ -69,65 +37,78 @@ pub fn fetch_chain_info(
     let url = format!("{}/info", base_url);
     match transport.fetch(&url) {
         Err(_) => Err(DrandClientError::NotResponding),
-        Ok(body) => serde_json::from_str(&body).map_err(|_| InvalidChainInfo),
+        Ok(body) => serde_json::from_str(&body).map_err(|e| {
+            println!("{}", e);
+            InvalidChainInfo
+        }),
     }
 }
 
-impl<'a, B> DrandClient<'a, B>
-where
-    B: DeserializeOwned,
-{
-    pub fn latest_randomness(&self) -> Result<B, DrandClientError> {
+impl<'a, T: Transport> DrandClient<'a, T> {
+    pub fn latest_randomness(&self) -> Result<Beacon, DrandClientError> {
         self.fetch_beacon_tag("latest")
     }
 
-    pub fn randomness(&self, round_number: u64) -> Result<B, DrandClientError> {
+    pub fn randomness(&self, round_number: u64) -> Result<Beacon, DrandClientError> {
         if round_number == 0 {
             return Err(InvalidRound);
         }
         self.fetch_beacon_tag(&format!("{}", round_number))
```

Comment:
You really don't like the format inlined syntax :P

Reply:
literally didn't know it existed x)